api/layer/system_object.go

```diff
@@ -35,12 +41,12 @@ func (n *layer) PutLockInfo(ctx context.Context, objVersion *ObjectVersion, newL
 		lockInfo = &data.LockInfo{}
 	}
 
-	if newLock.Retention != nil {
+	if p.NewLock.Retention != nil {
```

Comment:
Can we keep `newLock` variable? Define it as `newLock := p.NewLock` in the beginning of the function like `cnrID`. But it's up to you.